### DISCUSSION TITLE
IPv6 DHCP-PD  -  use a network identifier for interfaces possible

### DISCUSSION THREAD

**Discussion by rcmcronny** [rsmarples/dhcpcd#196] [Q&A] [answered]
Hi,

i am in the process on build the framework on my link to use ipv6 on my VDSL link, ipv6 works, the dhcp-pd works also and i can add a /64 to my primary interface lan facing.

The documentation is mostly on the internet for the wide-dhcp clients, the dhcpcd things are small and only for a "normal" case.
I have some more vlans and would like to split my dhcp-pd /56 to these all as a defined /64.
On the opnsense, i can set an "IPv6 Prefix ID"  0x1 or similar.

It seems the wide-dhcp client can that also with for example:
id-assoc pd 0 {   prefix-interface vlan10 {     sla-id 10;     sla-len 8;   }; 

But i use dhcpcd and would like to do that here, the man page and examples do not give many infos on this.

My config for my ppp0 interface looks like this:
_==============
allowinterfaces ppp0
duid
noipv6rs
waitip 6
ipv6only
interface ppp0
ipv6rs
iaid 1
ia_pd 1 br1
==============_

br1 is my lan facing bridge (i have also more interfaces (br6 for example).  
ppp0 is my internet facing interface

So perhaps someone can give me a hint on that here ?

Many Thanks,
Ronny

**Comment by rsmarples** [comment]
From dhcpcd.conf(5) man page.

```
ia_pd [iaid [/ prefix / prefix_len] [interface [/ sla_id [/ prefix_len [/suffix]]]]]
```

So the stanza you want to match your example is this:
```
ia_pd 0 vlan10/10/8/64
```
In wide-dhcp6 you specify the sla_len, but dhcpcd requires the prefix_length instead and attempts to fit the sla in. Here, I've assumed a /64.

Does this help?

**Comment by rcmcronny** [comment, marked as answer]
Hi @rsmarples 
many thanks, did not find it in my dhcpcd man page, but now i did :)

But i'm buffling with a working config:

I change my config for my br1 device from   
_ia_pd 1 br1_
to 
_ia_pd 1 br1/1/8/64_
so it should use build the /64 with an id 1, but a run shows 

_...
ppp0: delegated prefix 2001:aaa:bbb:1f00::/56
br1: invalid prefix 2001:aaa:bbb:1f00::/56 + 1/8: Numerical result out of range
..._

Config is for this: 
_==============
allowinterfaces ppp0
duid
noipv6rs
waitip 6
ipv6only
interface ppp0
ipv6rs
iaid 1
ia_pd 1 br1/1/8/64
==============_

I did find a solution for me:

ia_pd 1 br1/1 br6/6 net4.2/2 net4.4/4

this assigns perfectly the id 1 to br1, id 6 to br6 , id 2 to net4.2 and id4 to net4.4  
lesser is more it seems in my case :) 

Many thanks for your hint on this 👍
